app/src/main/java/com/example/androidthings/testapp/singleton/SerialSingleton.kt

```diff
@@ -0,0 +1,56 @@
+package com.example.androidthings.testapp.singleton
+
+import android.util.Log
+import com.google.android.things.pio.PeripheralManager
+import com.google.android.things.pio.UartDevice
+import utils.TAG
+import java.io.IOException
+
+object SerialSingleton {
+
+    private var uart: UartDevice = PeripheralManager.getInstance().openUartDevice("UART0")
+
+    init {
+        uart.apply {
+            // Configure the UART port
+            setBaudrate(115200)
+            setDataSize(8)
+            setParity(UartDevice.PARITY_NONE)
+            setStopBits(1)
+            setHardwareFlowControl(UartDevice.HW_FLOW_CONTROL_NONE)
+        }
+    }
+
+    /** Convert received byte array to String :
+     *      @filter : get only if incoming byte
+     *      @joinToString : add byte to string together with no separator
+     *                      convert to character and string
+     **/
+    private fun ByteArray.toReadableString() = filter{it>0.toByte()}
```

Comment:
it>0.toByte()

ça fait quoi de beau ?

Reply:
"it" est une variable qui récupère la valeur de la variable (ByteArray) qui l'appelle. "it>0" dans la fonction "filter" me permet de ne pas lire les bytes à 0. (stream null)